### PR TITLE
feat: add `updates.jenkins.io` storage account, redis database and CNAME record

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: https://github.com/jenkins-infra/helpdesk/issues/2649

### CODE REVIEW

**Comment by lemeurherve** [comment]
> Plan: 4 to add, 0 to change, 0 to destroy.

File: updates.jenkins.io.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Storage account
+resource "azurerm_resource_group" "updates_jenkins_io" {
+  name     = "updates-jenkins-io"
+  location = var.location
+  tags     = local.default_tags
+}
+
+resource "azurerm_storage_account" "updates_jenkins_io_files" {
+  name                          = "updatesjenkinsiofiles"
```
Comment (dduportal) [inline updates.jenkins.io.tf:9]:
```suggestion
  name                          = "updatesjenkinsio"
```

=> A storage account can have different services: files, blob storages, etc.

File: updates.jenkins.io.tf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  resource_group_name           = azurerm_resource_group.updates_jenkins_io.name
+  location                      = azurerm_resource_group.updates_jenkins_io.location
+  account_tier                  = "Standard"
+  account_replication_type      = "GRS"    # recommended for backups
```
Comment (dduportal) [inline updates.jenkins.io.tf:13]:
```suggestion
  account_replication_type      = "LRS"    # recommended for backups
```

Note sure why you would want to use GRS? https://learn.microsoft.com/en-us/azure/storage/common/storage-redundancy states that LRS are more cost effective and it's not clear why we would want to georeplicate?

File: updates.jenkins.io.tf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  tags = local.default_tags
+}
+
+output "updates_jenkins_io_files_primary_access_key" {
+  sensitive = true
+  value     = azurerm_storage_account.updates_jenkins_io_files.primary_access_key
+}
+
+# Redis database
+resource "azurerm_redis_cache" "updates_jenkins_io_database" {
```
Comment (dduportal) [inline updates.jenkins.io.tf:26]:
```suggestion
resource "azurerm_redis_cache" "updates_jenkins_io" {
```

The name "database" is already implied by the fact it is a Redis resource

**Review by dduportal — CHANGES_REQUESTED** [review]
First pass of review.

- Naming is not blocking
- Using `GRS` sounds like a problem at first sight though

**Pull request merged**